router/batchrouter/asyncdestinationmanager/snowpipestreaming/apiadapter.go

```diff
@@ -0,0 +1,124 @@
+package snowpipestreaming
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/snowpipestreaming/internal/model"
+)
+
+const (
+	createChannelAPI = "create_channel"
+	deleteChannelAPI = "delete_channel"
+	insertAPI        = "insert"
+	statusAPI        = "status"
+)
+
+func newApiAdapter(
+	logger logger.Logger,
+	statsFactory stats.Stats,
+	api api,
+	destination *backendconfig.DestinationT,
+) api {
+	return &apiAdapter{
+		logger:       logger,
+		statsFactory: statsFactory,
+		api:          api,
+		destination:  destination,
+	}
+}
+
+func (a *apiAdapter) defaultTags(apiName string) stats.Tags {
+	return stats.Tags{
+		"module":        "batch_router",
+		"workspaceId":   a.destination.WorkspaceID,
+		"destType":      a.destination.DestinationDefinition.Name,
+		"destinationId": a.destination.ID,
+		"api":           apiName,
+	}
+}
+
+func (a *apiAdapter) CreateChannel(ctx context.Context, req *model.CreateChannelRequest) (*model.ChannelResponse, error) {
+	a.logger.Infon("Creating channel",
+		logger.NewStringField("rudderIdentifier", req.RudderIdentifier),
+		logger.NewStringField("partition", req.Partition),
+		logger.NewStringField("database", req.TableConfig.Database),
+		logger.NewStringField("namespace", req.TableConfig.Schema),
+		logger.NewStringField("table", req.TableConfig.Table),
+	)
+
+	tags := a.defaultTags(createChannelAPI)
+	defer a.recordDuration(tags)()
+
+	resp, err := a.api.CreateChannel(ctx, req)
+	if err != nil {
+		tags["success"] = "false"
+		return nil, err
+	}
+	tags["success"] = strconv.FormatBool(resp.Success)
+	tags["code"] = resp.Code
+	return resp, nil
+}
```

Comment:
Timeout is already defined on the HTTP client that makes the request. Also, error enhancements might not be required since it's just an adapter that logs, sends stats, and returns the error as it is.